deploy/ControllerInit.sol

```diff
@@ -67,18 +71,45 @@ library MainnetControllerInit {
         IALMProxy   almProxy   = IALMProxy(controllerInst.almProxy);
         IRateLimits rateLimits = IRateLimits(controllerInst.rateLimits);
 
-        IMainnetControllerLike controller = IMainnetControllerLike(controllerInst.controller);
+        MainnetController controller = MainnetController(controllerInst.controller);
 
-        // Step 1: Configure ACL permissions for controller and almProxy
+        // Step 1: Perform sanity checks
+
+        require(almProxy.hasRole(DEFAULT_ADMIN_ROLE, params.admin)   == true, "MainnetControllerInit/incorrect-admin-almProxy");
+        require(controller.hasRole(DEFAULT_ADMIN_ROLE, params.admin) == true, "MainnetControllerInit/incorrect-admin-controller");
+        require(rateLimits.hasRole(DEFAULT_ADMIN_ROLE, params.admin) == true, "MainnetControllerInit/incorrect-admin-rateLimits");
```

Comment:
I would move the `almProxy` and `rateLimits` checks into the full version because we are not initializing those.

Reply:
Moved and updated tests